### ISSUE TITLE
100% CPU usage and stuck in reading an mp3 file

### ISSUE THREAD

**Issue by leofiore** [alsaplayer/alsaplayer#14]
_System:_
Debian GNU/Linux Jessie (8.3)
alsaplayer 0.99.81

_Steps to reproduce:_
download the ![attached file png](https://cloud.githubusercontent.com/assets/655964/13757650/ce85bb76-ea25-11e5-9776-bead849c5e73.png), rename it as .mp3 and open it via command line or adding it to the playlist manager

_Actual result:_
the file is never played, nor added to the playlist; instead, alsaplayer is stucked, probably trying reading the file.

[this log](https://github.com/alsaplayer/alsaplayer/files/172749/strace.txt) shows that at a certain point the player is continuosly looping through a `_llseek(6, 4096, [4096], SEEK_SET)      = 0`, until SIGINT is called.

_Expected result:_
file is added to the playlist and ready to be played

I'll try to grab more information later.

**Comment by leofiore** [comment]
Hello,

i collected more informations about this bug.

First of all i forgot to mention the system i was using is 32bit. I tested it on debian sid 64-bit and ubuntu trusty 64 bit, and everything works good. I then created a new 32 bit virtual machine with a fresh installation of debian jessie; here the bug still occurs.

After a few investigation on code i reached the conclusion that the problem lies on `ftell` usage in [parse_id3 function](https://github.com/alsaplayer/alsaplayer/blob/master/inpu/mad/mad_engine.c#L574). Basically, the stop condition is never reached because the `header_size + 10` value is never reached.
